CMakeLists.txt

```diff
@@ -266,7 +279,7 @@ IF(SECURITY_HARDENED AND NOT WITH_ASAN AND NOT WITH_UBSAN AND NOT WITH_TSAN AND
 ENDIF()
 
 OPTION(WITH_DBUG_TRACE "Enable DBUG_ENTER()/DBUG_RETURN()/DBUG_PRINT()" ON)
-IF(WITH_DBUG_TRACE)
+IF(WITH_DBUG_TRACE AND NOT WITH_MSAN)
```

Comment:
I think instead of doing this, it could be better to set the default value of `WITH_DBUG_TRACE` to `OFF`, while updating the `BUILD/` scripts so that they will explicitly specify `WITH_DBUG_TRACE=ON`. In any case, I think that this should be done in a separate ticket.

I understood that all `CMAKE_BUILD_TYPE=Debug` builds on https://buildbot.mariadb.org currently specify `WITH_DBUG_TRACE=OFF` in order to generate less output and avoid rather significant overhead. If we want to avoid doing that, we should disable `WITH_DBUG_TRACE` by default in CMake scripts. But, then some developer who values that facility could complain, or worse, re-enable it by default again without understanding the implications.